test_integration.py

```
@@ -0,0 +1,174 @@
+        status = manager.get_status()
+        assert status["total_files"] == 2
+        assert status["total_chunks"] == 80
+        assert status["percentage"] == 100.0
+        print(f"  ✓ Status: {status['indexed_files']}/{status['total_files']} files, {status['total_chunks']} chunks")
+
+        # Test 3: Test duplicate detection
+        print("\n3. Testing duplicate detection:")
+        state = manager.read_state()
+        normalized_path = UnifiedStateManager.normalize_path("/Users/ramakrishnanannaswamy/.claude/projects/test1.jsonl")
+        is_duplicate = normalized_path in state["files"]
+        assert is_duplicate == True
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `is_duplicate:` for truth checks

Replace with `is_duplicate`

(E712)